### PR TITLE
Feat(eos_designs,eos_cli_config_gen): Add support for VPN import pruning

### LINKED ISSUE #1172: Add support for VPN import prunning
### Enhancement summary

Implement VPN import prunning to not import VPN routes in BGP table on devices where VPNs are not configured.

### Which component of AVD is impacted

eos_designs, eos_cli_config_gen

### Use case example

Large deployments to not overconsume HW resoruces.

### Describe the solution you would like

Implement following CLI:

```
(config-router-bgp)#address-family evpn
(config-router-bgp-af)#route import match-failure action discard
```

### Describe alternatives you have considered

N/A

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by titom73**
## Change Summary

Add support for VPN import prunning for EVPN deployments.

## Related Issue(s)

Fixes #1172 

## Component(s) name

- `arista.avd.eos_cli_config_gen`
- `arista.avd.eos_designs`

## Proposed changes

__CLI Config Gen__

```yaml
router_bgp:
  address_family_evpn:
    route:
      import_match_failure_action: discard
```

__EOS Designs__

```yaml
# VPN import prunning
evpn_import_pruning: true
```

## How to test

Refer to molecule

```
# With no VPN Pruning - Dynamic VLAN is created for non-existing VRF
EAPI-BL01A#show vxlan vni
VNI to VLAN Mapping for Vxlan1
VNI         VLAN       Source       Interface       802.1Q Tag
----------- ---------- ------------ --------------- ----------
10110       110        static       Vxlan1          110
20201       201        static       Vxlan1          201

VNI to dynamic VLAN Mapping for Vxlan1
VNI       VLAN       VRF                      Source
--------- ---------- ------------------------ ------------
11        1198       TENANT_A_PROJECT01       evpn
13        1197                                evpn


# With VPN Pruning - Dynamic VLAN is created only for existing VRF
EAPI-BL01A#show vxlan vni
VNI to VLAN Mapping for Vxlan1
VNI         VLAN       Source       Interface       802.1Q Tag
----------- ---------- ------------ --------------- ----------
10110       110        static       Vxlan1          110
20201       201        static       Vxlan1          201

VNI to dynamic VLAN Mapping for Vxlan1
VNI       VLAN       VRF                      Source
--------- ---------- ------------------------ ------------
11        1198       TENANT_A_PROJECT01       evpn

EAPI-BL01A#
```

### User Checklist

- [x] eos_cli_config_gen implementation
- [x] eos_cli_config_gen molecule update
- [x] eos_designs implementation
- [x] eos_designs molecule update

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
Looks good. I took a look at the CLI, and we could flatten this a lot, since there are no alternative keywords:
```cli
ld001(config-router-bgp-af)#route ?
  import  Configure the importing of VPN routes
  type    NLRI route type

ld001(config-router-bgp-af)#route import ?
  match-failure  Configure how to handle VPN routes with no matching VRF import Route Targets

ld001(config-router-bgp-af)#route import match-failure ?
  action  Configure the action to perform

ld001(config-router-bgp-af)#route import match-failure action ?
  discard  Discard the route

ld001(config-router-bgp-af)#route import match-failure action discard ?
  <cr>  
```
So it _could_ be flattened to `route.import_match_failure_action: discard`.

**Comment by titom73** [comment]
Thanks for reviewing this PR. I have updated `eos_cli_config_gen` as per your comment

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2 [hunk 1]
```diff
@@ -156,6 +156,12 @@
 ### Router BGP EVPN Address Family
 {%     if router_bgp.address_family_evpn.evpn_hostflap_detection is arista.avd.defined %}
 
+{%         if router_bgp.address_family_evpn.route.import_match_failure_action is arista.avd.defined('discard') %}
+- VPN import prunning is __enabled__
+{%         else %}
+- VPN import prunning is disbaled
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2]: There is a typo here, "disabled".

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2 [hunk 2]
```diff
@@ -156,6 +156,12 @@
 ### Router BGP EVPN Address Family
 {%     if router_bgp.address_family_evpn.evpn_hostflap_detection is arista.avd.defined %}
 
+{%         if router_bgp.address_family_evpn.route.import_match_failure_action is arista.avd.defined('discard') %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2]: I don't think it is on purpose, that you enclose this `if` inside the `evpn_hostflap_detection`. So move this above the `if` in line 157, and then add a new `if` around this section (including 1 leading empty line) `if router_bgp.address_family_evpn.route.import_match_failure_action is arista.avd.defined`. 
This extra `if` serves the purpose of not changing documentation for all devices where this feature is not specifically enabled or disabled.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

**Review by ClausHolbechArista — APPROVED** [review]
Updated a typo and refreshed artifacts.
LGTM!

**Review by onurgashi — APPROVED** [review]

**Pull request merged**